### PR TITLE
Added a test for the randomBid() function #86

### PR DESCRIPTION
**Pull request by griffobeid**
## Description
Added a test in the simulation.vehicles.spec.js for the randomBid() function

## Related Issue
https://github.com/DAVFoundation/missioncontrol/issues/86

## Motivation and Context

The change is required to test that the randomBid() function is outputting an object with a price containing a float. 

## How Has This Been Tested?
I tested my changes by running an "npm test" and all tests passed, including the newly added test. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: server/simulation/vehicles.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const randomBid = (origin, pickup, dropoff) => {
   const dropoffPoint = turf.point([parseFloat(dropoff.long), parseFloat(dropoff.lat)]);
   const distanceOriginToPickup = turf.distance(originPoint, pickupPoint);
   const distancePickupToDelivery = turf.distance(pickupPoint, dropoffPoint);
-  const price = (distanceOriginToPickup+distancePickupToDelivery).toFixed(2);
+  const price = parseFloat((distanceOriginToPickup+distancePickupToDelivery).toFixed(2));
```
Comment (TalAter) [inline server/simulation/vehicles.js]: I see the value of what you did here... but this task is about testing the current state of the app.

The change you introduced here changes the type of data the app returns and may have unintended effects here and in Missions. In fact, it will change the way prices are displayed in Missions' UI, which we do not want.

In other words, this is a *breaking change* and not something that can be introduced as part of another unrelated pull request. It needs to be considered and tested carefully and missions would need to be updated accordingly... just a part of the reality of developing open source software that is used by many different people. You need to make sure you do not accidentally break other people's software who may rely on that being a string (even if it shouldn't be).

But yes, this does seem like a good change and you uncovered an actual issue here.

What I suggest is that you create a separate pull request to Missions that expects to receive a float in the price field and calls `toFixed(2)` when it wants to display that value in the UI. Then make the change you suggested to this method in a separate pull request to Mission Control. Then finally we'll do this pull request here that only handles testing.

I know it sounds like a hassle, but I think it should take you less time to do it then it took me to write the above ;)

File: test/specs/simulation.vehicles.spec.js [hunk 2]
```diff
@@ -1,19 +1,37 @@
-const { generateRandomVehicles } = require('../../server/simulation/vehicles.js');
+const {generateRandomVehicles, randomBid} = require('../../server/simulation/vehicles.js');
```
Comment (TalAter) [inline test/specs/simulation.vehicles.spec.js]: There are a lot of formatting changes introduced in this file that have nothing to do with what you added.
My guess is one of your tools (prettier?) introduced them automatically.

When contributing to open source, try to follow the existing code style of the project you are contributing to (in this example it is spaces around the variable names in line 1), and only change the relevant code.

If you'd like to introduce further unrelated changes, you can do them in a separate pull request

File: test/specs/simulation.vehicles.spec.js [hunk 3]
```diff
@@ -1,19 +1,37 @@
...
   test('returns an array of objects specifically matching given props', () => {
-    expect(
-      Array.isArray(generateRandomVehicles(1, {}, 2000))
-    ).toBe(true);
+    expect(Array.isArray(generateRandomVehicles(1, {}, 2000))).toBe(true);
 
-    expect(
-      Object.keys(generateRandomVehicles(1, {}, 2000)[0]).sort()
-    ).toEqual(['id', 'model', 'icon', 'coords', 'rating', 'missions_completed', 'missions_completed_7_days', 'status'].sort());
+    expect(Object.keys(generateRandomVehicles(1, {}, 2000)[0]).sort()).toEqual([
```
Comment (TalAter) [inline test/specs/simulation.vehicles.spec.js]: Again, all of these changes are unrelated.

File: test/specs/simulation.vehicles.spec.js [hunk 4]
```diff
@@ -1,19 +1,37 @@
...
+      'coords',
+      'rating',
+      'missions_completed',
+      'missions_completed_7_days',
+      'status'
+    ].sort());
+  });
+});
+
+describe('randomBid()', () => {
```
Comment (TalAter) [inline test/specs/simulation.vehicles.spec.js]: May I suggest improving the first part for conciseness and breaking the test itself for readability?

```javascript
describe('randomBid()', () => {
  const getRandomCoords = randomCoords({
    coords: {
      lat: 1,
      long: 1
    },
    radius: 1000
  });

  test('returns an object with a price containing a float', () => {
    const bid = randomBid(getRandomCoords(), getRandomCoords(), getRandomCoords());
    expect(typeof bid.price).toBe('number');
  });
});
```

File: server/simulation/vehicles.js (hunk 1 above)
Comment (griffobeid) [reply server/simulation/vehicles.js]: I had to use an alternative solution for trimming the decimals, but here is the pull request: https://github.com/DAVFoundation/missions/pull/40 😀

**Comment by TalAter** [comment]
Looking good 👍 

I'll merge as soon as the missions one is ready as well.
